// react-elfsight-widget/dist/react-elfsight-widget.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);

var PLATFORM_URL = 'https://static.elfsight.com/platform/platform.js';
function useLoadPlatform() {
  React.useEffect(function () {
    if (!isPlatformLoaded()) {
      loadPlatform();
    }
  }, []);
}

function isPlatformLoaded() {
  return isPlatformInitialized() || hasPlatformScript();
}

function isPlatformInitialized() {
  return 'eapps' in window;
}

function hasPlatformScript() {
  return !!document.querySelector("script[src=\"" + PLATFORM_URL + "\"]");
}

function loadPlatform() {
  var scriptElement = document.createElement('script');
  scriptElement.src = PLATFORM_URL;
  scriptElement.defer = true;
  document.head.appendChild(scriptElement);
}

function _ElfsightWidget(_ref) {
  var widgetID = _ref.widgetID;
  useLoadPlatform();
  return React__default.createElement("div", {
    className: "elfsight-app-" + widgetID
  });
}

var ElfsightWidget = /*#__PURE__*/React.memo(_ElfsightWidget);

exports.ElfsightWidget = ElfsightWidget;
